ui/src/components/auth/ResetPasswordContainer.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import axios from 'axios';
+import {
+  Box,
+  Button,
+  Stack,
+  TextField,
+  Typography,
+  Snackbar,
+  Alert,
+  AlertColor,
+} from '@mui/material';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { ResetPasswordFields } from './helpers/form.types';
+import { formSpacing, formWidth } from './helpers/constants';
+import { useLocation } from 'react-router-dom';
+import { stringValidator, passwordValidator } from './helpers/zod.validators';
+
+// zod validation
+const resetPwdValidationSchema = z
+  .object({
+    password: passwordValidator,
+    confirmPassword: stringValidator,
+  })
+  .refine(
+    (values) => {
+      return values.password === values.confirmPassword;
+    },
+    {
+      message: 'The passwords do not match',
+      path: ['confirmPassword'],
+    }
+  );
+
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
+
+const ResetPasswordContainer: React.FC = (props: any) => {
+  const { history } = props;
+
+  const query = useQuery();
+
+  const [open, setOpen] = useState<boolean>(false);
+  const [status, setStatus] = useState<AlertColor | undefined>();
+  const [message, setMessage] = useState<string>('');
+
+  const handleClose = (event?: React.SyntheticEvent | Event, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+
+    setOpen(false);
+  };
+
+  const { register, handleSubmit, formState } = useForm<ResetPasswordFields>({
+    resolver: zodResolver(resetPwdValidationSchema),
+  });
+
+  const { errors } = formState;
+
+  const handleResetPassword: SubmitHandler<ResetPasswordFields> = async (data) => {
+    try {
+      const token = query.get('token');
+      const email = query.get('email');
+
+      const payload = await axios.post('/api/auth/reset-password', {
+        email: email,
+        token: token,
+        password: data.password ?? '',
+      });
+
+      if (payload.status === 200) {
+        setStatus('success');
+        setMessage('Password successfully reset! Please login again.');
+        setTimeout(() => {
+          history.push('/login');
+        }, 3000);
```

Comment:
Is 3 second too fast? Lol

Reply:
LOLOL

Reply:
i think it's okay laa, i would prefer responsiveness than long feedback hehe